apis/r/configure

```diff
@@ -10,8 +10,14 @@ if test -z "${R_HOME}"; then
 fi
 
 ### for single-cell-data/apis/r repo, consider completing sources
-if test -f src/lib/libtiledbsoma.a; then
-    echo "** static library present"
+pkg-config --exists tiledbsoma
+if [ $? -eq 0 ]; then
```

Comment:
If we order it this way, R will first try to find the shared library.  That we now decided should not have the 'R_BUILD' enabled.  This risks breaking R builds.